### DISCUSSION TITLE
The easiest way to move an offset

### DISCUSSION THREAD

**Discussion by Serg046** [jbevain/cecil#784] [Q&A] [answered]
As answered [here](https://stackoverflow.com/a/7267875/3561198), the lib takes care about br instructions so that they point to the right places. For example, I have the following
```
IL_0001 br IL_0003
...
IL_0003 call ...
```
and use InsertBefore(IL_0003) to put two additional instructions:
```
IL_0001 br IL_0003
...
IL_0000 nop
IL_0000 nop
IL_0003 call ...
```
So `br` still points to `ret`. Usually this is what you want but in my case I need to point to the first `nop` because it is additional work to configure a valid state for `call`. Of course, I can just look for all `br`-like related instructions and update them but maybe you know a better way that you can advise here. E.g. I can imagine using of Replace(...) so that I have something like that
```
IL_0001 br IL_0003
...
IL_0003 nop #replaced using Replace(...)
IL_0000 nop
IL_0000 call ...
```
But it is not convenient too.

**Comment by ltrzesniewski** [comment, marked as answer]
I suppose the easiest way would be to *convert* the `call` instruction into a `nop` by changing the opcode on the `Instruction` instance and clearing its operand, then inserting the second `nop` and a `call` instruction created from scratch afterwards. That way existing references would be preserved, but would now point to the first `nop`.

Maybe that's not very intuitive, but at least it won't require you to update the references (which [is a bit tricky to get right](https://github.com/jbevain/cecil/issues/182#issuecomment-858385597)).

**Reply by Serg046** [reply]
Hmm, thank you! I think it is convenient enough 👍
